common/rpc/interceptor/telemetry.go

```diff
@@ -110,11 +111,18 @@ func (ti *TelemetryInterceptor) Intercept(
 	timer := metricsScope.StartTimer(metrics.ServiceLatency)
 	defer timer.Stop()
 
+	timerNoUser := metricsScope.StartTimer(metrics.ServiceLatencyNoUser)
+
 	resp, err := handler(ctx, req)
+	propagated := metrics.GetPropagatedMetricsFromContext(ctx)
 	if err != nil {
 		ti.handleError(scope, methodName, err)
 		return nil, err
 	}
+	if val, ok := propagated[metrics.UserInducedLatency]; ok {
+		timerNoUser.Substract(time.Duration(val))
+		defer timerNoUser.Stop()
+	}
```

Comment:
not sure i understand the PR correctly, but how about the following
```go

overallLatency := ...

nonUserLatency := overallLatency
for _, Item := range extractUserLatency(ctx) {
    nonUserLatency -= Item.GetLatency()
}

// export overallLatency
// export nonUserLatency
```

Reply:
so in the future we have have easy access to each individual item

Reply:
In essence that's what's happening here. I'll cleanup the code.